pkg/skaffold/runner/util/util.go

```diff
@@ -38,11 +38,13 @@ func GetAllPodNamespaces(configNamespace string, pipelines []latest.Pipeline) ([
 			return nil, fmt.Errorf("getting k8s configuration: %w", err)
 		}
 
+		// The empty string is here to ensure that the File Sync feature
+		// works with multi-namespace environments
+		nsMap[""] = true
```

Comment:
the reason why we we are against "" in nsMap is because not everyone has permissions to fetch pods and objects in all namespaces. 
With empty namespace, k8s api will try to fetch pods, monitors etc in all namespaces.
To avoid getting a permission issue, we only add "" when no namespace is present.

Reply:
Ah right, I see. My assumption was that skaffold is only used with local clusters (`minikube` and similar), but now your comment implies people use it with shared clusters too. So, in that case I see how this would cause problems.

This is mostly an issue for kustomize  and kubectl deployments, since they're simply "glide-through" skaffold without any processing.

Could I instead peek at `.metadata.namespace` of the kustomize-compiled resources? I know could be made possible, but I'm not sure if that's desirable? In that case it would work in a similar fashion to the Helm namespace heuristic.

Reply:
Looking more into this, we provide an `UpdateNamespace` method which updates the namespaces with deployed resources.
https://github.com/GoogleContainerTools/skaffold/blob/5bf0323473ce7610a6cde65ce1f882f886e029ea/pkg/skaffold/runner/runcontext/context.go#L244

I submitted a PR recently where `PodWatcher.Start()` uses updated namespace. 
I am wondering if that fixed this issue. 
https://github.com/GoogleContainerTools/skaffold/pull/5487

Can you verify this on the latest version of skaffold?